### PR TITLE
[react-relay] Add plural fragment refs and handle connection data well.

### PR DESCRIPTION
**Pull request by alloy**
Will comment inline. All other changes are just prettier changes, especially in the tests which we hadn’t been prettifying previously.

### CODE REVIEW

File: types/react-relay/index.d.ts [hunk 1]
```diff
@@ -9,6 +9,10 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
+// Prettified with:
+// $ prettier --parser typescript --tab-width 4 --semi --trailing-comma es5 --write --print-width 120 \
+//   types/{react-relay,relay-runtime}/{,*}/*.ts*
```
Comment (alloy) [inline types/react-relay/index.d.ts]: Added the prettier command I use to prettify all related files.

**Comment by alloy** [comment]
Interesting that @kastermester didn’t get a ping for a review 🤔

**Review by kastermester — APPROVED** [review]
This looks fine to me :)

I'm a bit curious of which changes you'd want to make to the `ConnectionData` interface?

**Comment by alloy** [comment]
@kastermester Can you expand a bit on what you mean? The changes I’ve made in this PR or in the future related to fragment references?

**Comment by alloy** [comment]
(Merging after the approved review)

File: types/react-relay/index.d.ts [hunk 2]
```diff
@@ -113,8 +122,8 @@ export interface PageInfo {
     startCursor: string | undefined | null;
 }
 export interface ConnectionData {
-    edges?: any[];
-    pageInfo?: PageInfo;
+    edges?: ReadonlyArray<any>;
+    pageInfo?: Partial<PageInfo>;
```
[the remark this answers is not in the document]
Comment (kastermester) [reply types/react-relay/index.d.ts]: I was talking about this comment here. What kind of flexibility is it you wish to add?

[the remark this answers is not in the document]
Comment (alloy) [reply types/react-relay/index.d.ts]: Ah, so this makes it so both `ReadonlyArray` and `Array` edges are valid (most important is the former, as that’s what our plugin outputs). Additionally users may only need a few page info fields, so requiring them all to recognize the data as a valid connection is encouraging overfetching.

[the remark this answers is not in the document]
Comment (kastermester) [reply types/react-relay/index.d.ts]: I don't think this type requires that much (possibly even not enough). I think the only real restriction here is that `edges` cannot be `null`. Other than that it seems to me like it's all accepted here?

**Pull request merged**